docs/auth.md

```diff
@@ -4,35 +4,224 @@ linkTitle: "Authentication"
 weight: 7
 ---
 -->
-# Authentication
+# Authentication at Run Time
+
+This document describes how Tekton handles authentication when executing
+`TaskRuns` and `PipelineRuns`. Since authentication concepts and processes
+apply to both of those entities in the same manner, this document collectively
+refers to `TaskRuns` and `PipelineRuns` as `Runs` for the sake of brevity.
+
+- [Overview](#overview)
+- [Understanding credential selection](#understanding-credential-selection)
+- [Authenticating as a non-root user](#authenticating-as-a-non-root-user)
+- [Limiting `Secret` access to specific `Steps`](#limiting-secret-access-to-specific-steps)
+- [Configuring authentication for Git](#configuring-authentication-for-git)
+  - [Configuring `basic-auth` authentication for Git](#configuring-basic-auth-authentication-for-git)
+  - [Configuring `ssh-auth` authentication for Git](#configuring-ssh-auth-authentication-for-git)
+  - [Using a custom port for SSH authentication](#using-a-custom-port-for-ssh-authentication)
+  - [Using SSH authentication in `git` type `Tasks`](#using-ssh-authentication-in-git-type-tasks)
+- [Configuring authentication for Docker](#configuring-authentication-for-docker)
+  - [Configuring `basic-auth` authentication for Docker](#configuring-basic-auth-authentication-for-docker)
+  - [Configuring `docker*` authentication for Docker](#configuring-docker-authentication-for-docker)
+- [Technical reference](#technical-reference)
+  - [`basic-auth` for Git](#basic-auth-for-git)
+  - [`ssh-auth` for Git](#ssh-auth-for-git)
+  - [`basic-auth` for Docker](#basic-auth-for-docker)
+
+## Overview
+
+Tekton supports authentication via the Kubernetes first-class `Secret` types listed below.
+A `Run` gains access to these `Secrets` through its associated `ServiceAccount`.
+
+<table>
+	<thead>
+		<th>Git</th>
+		<th>Docker</th>
+	</thead>
+	<tbody>
+		<tr>
+			<td><code>kubernetes.io/basic-auth</code><br>
+				<code>kubernetes.io/ssh-auth</code>
+			</td>
+			<td><code>kubernetes.io/basic-auth</code><br>
+				<code>kubernetes.io/dockercfg</code><br>
+				<code>kubernetes.io/dockerconfigjson</code>
+			</td>
+	</tbody>
+</table>	
+
+Since neither Git nor Docker can consume these `Secrets` in their native format, Tekton must first
+convert them as follows:
+
+- For Git, convert into a `.gitconfig` file.
+- For Docker, convert into a `~/.docker/config.json` file.
+
+Additionally, even though each of these entities supports multiple credentials for multiple
+domains, Tekton must first blend those credentials into a single canonical keyring.
+
+To consume these `Secrets`, Tekton performs credential initialization within every `Pod` it instantiates,
+before retrieving any associated `PipelineResources`. During credential initialization, Tekton accesses
```

Comment:
nit: this is not related to `PipelineResources`. Even without any `PipelineResource` the *credential initialijation* will happen.

Reply:
Changed to "[...] before executing any `Steps` in the `Run`."